src/lib/features/feature-toggle/feature-toggle-store.ts

```diff
@@ -718,6 +722,40 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
 
         return result?.potentially_stale ?? false;
     }
+
+    async setCreatedByUserId(batchSize: number): Promise<void> {
+        const EVENTS_TABLE = 'events';
+        const USERS_TABLE = 'users';
+        const API_TOKEN_TABLE = 'api_tokens';
+
+        if (!this.flagResolver.isEnabled('createdByUserIdDataMigration')) {
+            return;
+        }
+        const toUpdate = await this.db(`${TABLE} as f`)
+            .joinRaw(`JOIN ${EVENTS_TABLE} AS ev ON ev.feature_name = f.name`)
+            .joinRaw(
+                `LEFT OUTER JOIN ${USERS_TABLE} AS u on ev.created_by = u.username OR ev.created_by = u.email`,
+            )
+            .joinRaw(
+                `LEFT OUTER JOIN ${API_TOKEN_TABLE} AS t on ev.created_by = t.username`,
+            )
+            .whereRaw(
+                `f.created_by_user_id IS null AND ev.type = 'feature-created'`,
```

Comment:
.whereRaw('u.id is not null or t.username is not null')
potential improvement so we don't get the same 100 every time